libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridSplineToGrid.cpp

```diff
@@ -0,0 +1,904 @@
+#include "MeshKernel/CurvilinearGrid/CurvilinearGridSplineToGrid.hpp"
+#include "MeshKernel/Constants.hpp"
+#include "MeshKernel/Utilities/LinearAlgebra.hpp"
+
+#include <algorithm>
+#include <iomanip>
+
+meshkernel::CurvilinearGrid meshkernel::CurvilinearGridSplineToGrid::Compute(const Splines& splines,
+                                                                             const CurvilinearParameters& curvilinearParameters) const
+{
+    if (splines.GetNumSplines() < 4)
+    {
+        throw ConstraintError("At least 4 splines are required to generate a grid, number of splines is {}", splines.GetNumSplines());
+    }
+
+    if (!CheckSplines(splines))
+    {
+        throw ConstraintError("One or more splines has one or fewer support points");
+    }
+
+    if (std::max(curvilinearParameters.m_refinement, curvilinearParameters.n_refinement) > MaximumRefinementFactor)
+    {
+        throw ConstraintError("Reduce the refinement factor to about 50 or less, currently n- and m-fac are: {} and {} respectively",
+                              curvilinearParameters.n_refinement, curvilinearParameters.m_refinement);
+    }
+
+    CurvilinearGrid grid(splines.m_projection);
+
+    // Make copy of spline because the may be changed
+    // i.e. the order may change and the number of spline points may increase.
+    Splines splinesCopy(splines);
+    VectorOfDoubleVectors splineIntersections(splines.GetNumSplines(), DoubleVector(splines.GetNumSplines(), 0.0));
+
+    UInt mRefinement = curvilinearParameters.m_refinement;
+    UInt nRefinement = curvilinearParameters.n_refinement;
+
+    // mn12 in Fortran.
+    VectorOfThreeInts splineInteraction(splines.GetNumSplines());
+    UInt numNSplines = 0;
+
+    ComputeSplineIntersections(splinesCopy, splineIntersections, numNSplines);
+    OrderSplines(splinesCopy, numNSplines, splineIntersections);
+    DetermineSplineOrientation(splinesCopy, numNSplines, splineIntersections, splineInteraction);
+    GenerateGrid(splinesCopy, splineIntersections, splineInteraction, numNSplines, mRefinement, nRefinement, grid);
```

Comment:
can you return the grid instead? can we use the move constructor for the curvilinear grid?

Reply:
It should already use the move constructor

Reply:
Add addition function to mode grid nodes